src/matrix/kaldi-matrix.cc

```diff
@@ -2717,6 +2717,51 @@ void MatrixBase<Real>::AddToRows(Real alpha, Real *const *dst) const {
   }
 }
 
+template<typename Real>
```

Comment:
Please don't add this function to the kaldi-matrix library-- it's too obscure to be something we want in such a core library.  Instead put the code directly on the other end of the 'if' statement in the cudamatrix function definition.